### PR TITLE
Feature/provisioning apply role

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

 * Added the ability to autoregister and assign a role to a device authorized in a provisioner
 * Added the ability to control whether or not a provisioner should be enforcing (i.e. ensuring all devices matching it are authorized with it)
 * Added the ability to sync the PID of devices authorized in a provisioner (only for Airwatch and JAMF)
 * Added VMware Airwatch support

# Impacts
Provisioning as a whole due to the various changes that were made

# Delete branch after merge
YES

# NEWS file entries
(REQUIRED, but may be optional...)
## New Features
 * Added the ability to autoregister and assign a role to a device authorized in a provisioner
 * Added the ability to control whether or not a provisioner should be enforcing (i.e. ensuring all devices matching it are authorized with it)
 * Added the ability to sync the PID of devices authorized in a provisioner (only for Airwatch and JAMF)
 * Added VMware Airwatch support

# UPGRADE file entries

None required, everything should be backward compatible

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/Provisioning/airwatch.pm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+has_field 'api_password' => (
+    type        => 'ObfuscatedText',
+    label       => 'API password',
+    required    => $TRUE,
+);
+
+has_field 'tenant_code' => (
+    type        => 'ObfuscatedText',
+    label       => 'API password',
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Provisioning/airwatch.pm]: Tenant code ?

File: lib/pf/provisioner/airwatch.pm [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+ 
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2019 Inverse inc.
```
Comment (fdurand) [inline lib/pf/provisioner/airwatch.pm]: 2020

File: lib/pf/provisioner/jamf.pm [hunk 3]
```diff
@@ -115,14 +124,20 @@ sub authorize {
     my $logger = get_logger;
 
     my ( $status, $device_type, $device_information ) = $self->get_device_information($mac);
-
+
```
Comment (fdurand) [inline lib/pf/provisioner/jamf.pm]: remove space

File: lib/pf/provisioner/jamf.pm [hunk 4]
```diff
@@ -250,12 +265,12 @@ sub parse_device_information {
     my $logger = get_logger;
 
     my $json = decode_json($device_information);
-
+
```
Comment (fdurand) [inline lib/pf/provisioner/jamf.pm]: remove space

Comment (julsemaan) [reply lib/pf/provisioner/jamf.pm]: I was trying to make some space for @lzammit returning back to work next week. Probably gained a few pounds in the last 5 weeks

**Comment by julsemaan** [comment]
This has been rebased and comments have been addressed

**Pull request merged**